pkg/apis/core/v1beta1/helper/shoot.go

```diff
@@ -748,3 +749,21 @@ func IsOneWorkerPoolLowerKubernetes134(controlPlaneVersion *semver.Version, work
 	}
 	return atLeastOnePoolLowerKubernetes134, nil
 }
+
+// GetShootGardenerOperations returns the Shoot's gardener operations specified in the operation annotation.
+func GetShootGardenerOperations(annotations map[string]string) []string {
+	return utils.SplitAndTrimString(annotations[v1beta1constants.GardenerOperation], v1beta1constants.GardenerOperationsSeparator)
+}
+
+// GetShootMaintenanceOperations returns the Shoot's maintenance operations specified in the operation annotation.
+func GetShootMaintenanceOperations(annotations map[string]string) []string {
+	return utils.SplitAndTrimString(annotations[v1beta1constants.GardenerMaintenanceOperation], v1beta1constants.GardenerOperationsSeparator)
+}
+
+// RemoveOperation removes listed operations from the operations slice.
+// Operations between the new length and the original length are replaced with empty strings.
```

Comment:
Thanks for removing the unnecessary `clone()` here, however, I feel like this new comment is not necessary and a bit misleading because that will happen only to the input argument and not the output which is an entirely new slice

I suggest something like this:
```suggestion
// RemoveOperation removes listed operations from the operations slice and returns a new slice that does not contain these operations. 
// Note that the input operations slice is modified.
```